src/main/java/net/neoforged/neoforge/network/PacketDistributor.java

```diff
@@ -8,81 +8,109 @@
-import java.util.function.Consumer;
-import java.util.function.Function;
-import java.util.function.Supplier;
 import net.minecraft.client.Minecraft;
+import net.minecraft.client.multiplayer.ClientPacketListener;
 import net.minecraft.network.protocol.Packet;
-import net.minecraft.network.protocol.PacketFlow;
 import net.minecraft.network.protocol.common.ClientboundCustomPayloadPacket;
-import net.minecraft.network.protocol.common.ServerboundCustomPayloadPacket;
 import net.minecraft.network.protocol.common.custom.CustomPacketPayload;
 import net.minecraft.network.protocol.game.ClientGamePacketListener;
 import net.minecraft.network.protocol.game.ClientboundBundlePacket;
 import net.minecraft.resources.ResourceKey;
 import net.minecraft.server.MinecraftServer;
 import net.minecraft.server.level.ServerChunkCache;
+import net.minecraft.server.level.ServerLevel;
 import net.minecraft.server.level.ServerPlayer;
 import net.minecraft.world.entity.Entity;
+import net.minecraft.world.level.ChunkPos;
 import net.minecraft.world.level.Level;
-import net.minecraft.world.level.chunk.LevelChunk;
 import net.neoforged.neoforge.server.ServerLifecycleHooks;
+import org.jetbrains.annotations.Nullable;
 
 /**
  * Means to distribute packets in various ways
  */
-public class PacketDistributor<T> {
+public final class PacketDistributor {
+    private PacketDistributor() {}
+
     /**
-     * Send to the player specified
-     * <br/>
-     * {@link #with(Object)} Player
+     * Send the given payload(s) to the server
      */
-    public static final PacketDistributor<ServerPlayer> PLAYER = new PacketDistributor<>(PacketDistributor::playerConsumer, PacketFlow.CLIENTBOUND);
+    public static void sendToServer(CustomPacketPayload... payloads) {
+        ClientPacketListener listener = Objects.requireNonNull(Minecraft.getInstance().getConnection());
+        for (CustomPacketPayload payload : payloads) {
+            listener.send(payload);
+        }
+    }
+
     /**
-     * Send to everyone in the dimension specified
-     * <br/>
-     * {@link #with(Object)} DimensionType
+     * Send the given payload(s) to the given player
      */
-    public static final PacketDistributor<ResourceKey<Level>> DIMENSION = new PacketDistributor<>(PacketDistributor::playerListDimConsumer, PacketFlow.CLIENTBOUND);
+    public static void sendToPlayer(ServerPlayer player, CustomPacketPayload... payloads) {
+        player.connection.send(makeClientboundPacket(payloads));
+    }
+
     /**
-     * Send to everyone near the {@link TargetPoint} specified
-     * <br/>
-     * {@link #with(Object)} TargetPoint
+     * Send the given payload(s) to all players in the given dimension
      */
-    public static final PacketDistributor<TargetPoint> NEAR = new PacketDistributor<>(PacketDistributor::playerListPointConsumer, PacketFlow.CLIENTBOUND);
+    public static void sendToPlayersInDimension(ResourceKey<Level> dimension, CustomPacketPayload... payloads) {
```

Comment:
Do we want this method (and some of the others below) to take a `ServerLevel` instead? Same comments for the other methods that use `getServer`. This would provide a bit more type safety at the call site, at the cost of a slightly more verbose interface.

Reply:
I have considered doing that:
- For the dimension target, I'd expect the caller to have the level anyway, so the level can replace the `ResourceKey`
- For the "near" target I would probably change the `TargetPoint` to hold a level instead of the `ResourceKey` (or we nuke `TargetPoint` and have the caller pass all args in directly since we aren't restricted to single-arg targets anymore)
- For the "all" target, I'm a bit divided since passing a level there could be a bit confusing. I would expect the caller to have a level available though
- For the chunk target I basically already did this since from my own experience you are more likely to have a level and a chunk position than to have the `LevelChunk`

Reply:
For the dimension target, I would switch to `ServerLevel`.  For near, I would pass the arguments directly and nuke `TargetPoint`.